### PR TITLE
UCT/SM: is_connected implementation for SM

### PR DESCRIPTION
**Pull request by shasson5**
## What
UCT/SM: is_connected implementation for SM

## Why ?
Support is_connected API

### CODE REVIEW

**Comment by gleon99** [comment]
@shasson5  please merge from master to resolve conflicts and prevent excessive diff.

File: src/uct/sm/mm/base/mm_ep.c [hunk 1]
```diff
@@ -71,6 +71,30 @@ uct_mm_ep_attach_remote_seg(uct_mm_ep_t *ep, uct_mm_seg_id_t seg_id,
...
+    uct_mm_iface_addr_t *mm_addr = (uct_mm_iface_addr_t*)params->iface_addr;
+    uct_mm_iface_t *iface;
+    uct_mm_md_t *md;
+
+    if (!uct_base_ep_is_connected(tl_ep, params)) {
+        return 0;
+    }
+
+    iface = ucs_derived_of(tl_ep->iface, uct_mm_iface_t);
+    md    = ucs_derived_of(iface->super.super.md, uct_mm_md_t);
```
Comment (yosefe) [inline src/uct/sm/mm/base/mm_ep.c:87]: init inline

File: src/uct/sm/mm/base/mm_ep.c [hunk 2]
```diff
@@ -71,6 +71,30 @@ uct_mm_ep_attach_remote_seg(uct_mm_ep_t *ep, uct_mm_seg_id_t seg_id,
     return UCS_OK;
 }
 
+int uct_mm_ep_is_connected(const uct_ep_h tl_ep,
+                           const uct_ep_is_connected_params_t *params)
+{
+    uct_mm_ep_t *ep              = ucs_derived_of(tl_ep, uct_mm_ep_t);
+    uct_mm_iface_addr_t *mm_addr = (uct_mm_iface_addr_t*)params->iface_addr;
```
Comment (yosefe) [inline src/uct/sm/mm/base/mm_ep.c:78]: must check flags before accessing iface_addr

File: src/uct/sm/scopy/cma/cma_ep.c [hunk 3]
```diff
@@ -84,6 +84,20 @@ uct_cma_ep_tx_error(uct_cma_ep_t *ep, const char *cma_call_name,
...
+                            const uct_ep_is_connected_params_t *params)
+{
+    uct_cma_ep_t *ep = ucs_derived_of(tl_ep, uct_cma_ep_t);
+    pid_t addr_pid;
+
+    if (!uct_base_ep_is_connected(tl_ep, params)) {
+        return 0;
+    }
+
+    addr_pid = *(pid_t*)params->iface_addr & ~UCT_CMA_IFACE_ADDR_FLAG_PID_NS;
```
Comment (yosefe) [inline src/uct/sm/scopy/cma/cma_ep.c:97]: extract common code with cma_ep constructor (get remote pid from cma iface address)

**Review by yosefe — APPROVED** [review]

File: src/uct/sm/mm/base/mm_ep.c [hunk 4]
```diff
@@ -71,6 +71,21 @@ uct_mm_ep_attach_remote_seg(uct_mm_ep_t *ep, uct_mm_seg_id_t seg_id,
     return UCS_OK;
 }
 
+int uct_mm_ep_is_connected(const uct_ep_h tl_ep,
+                           const uct_ep_is_connected_params_t *params)
+{
+    uct_mm_ep_t *ep = ucs_derived_of(tl_ep, uct_mm_ep_t);
```
Comment (gleon99) [inline src/uct/sm/mm/base/mm_ep.c:77]: `const`

**Review by gleon99 — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**